CRM/CiviAwards/BAO/AwardDetail.php

```diff
@@ -81,6 +81,10 @@ private static function validateParams(array &$params) {
    *   Parameters.
    */
   private static function validateDates(array $params) {
+    if (empty($params['start_date'])) {
+      throw new Exception("Award Start Date should not be empty");
+    }
+
```

Comment:
I think we should add a new method `validateMandatory` that will cater for all parameters that are mandatory and needs to be validated. 

Something like this

```php
private static function validateMandatory($params) {
    $mandatoryFields = [
      'start_date' => 'Start Date'
      'award_subtype' => 'Award Subtype',
      'award_manager' => 'Award Manager'
    ];

    foreach($mandatoryFields as $field => $label) {
      if (empty($params[$field])) {
        throw new Exception( "The {$label} field should not be empty",);
      }
    }
  }


```

If we need to make any other parameter mandatory, we'll just add to the array

Reply:
Good idea 💪 . Done here: https://github.com/compucorp/uk.co.compucorp.civiawards/pull/164/commits/5e2bb6395c6dbf6c97f47799681da2ed3f00d192